### PR TITLE
Add Range

### PR DESCRIPTION
**Pull request by jsrdxzw**

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	}
+	result := make([]int, -elementNum)
+	for i := 0; i < -elementNum; i++ {
+		result[i] = -i
+	}
+	return result
+}
+
+// RangeFrom creates an array of numbers from start with specified length.
+func RangeFrom(start, elementNum int) []int {
```
Comment (samber) [inline util.go:20]: `func RangeFrom[T constraints.Integer | constraints.Float](start T, elementNum int) []T {`

File: util.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	result := make([]int, -elementNum)
+	for i := 0; i < -elementNum; i++ {
+		result[i] = start - i
+	}
+	return result
+}
+
+// RangeOpen creates an array of numbers (positive and/or negative) progressing from start up to, but not including end.
+// step is default 1 when set to zero or can not reach end.
+func RangeOpen(start, end, step int) []int {
```
Comment (samber) [inline util.go:37]: `func RangeOpen[T constraints.Integer | constraints.Float](start T, end T, step T) []T {`

**Review by samber — CHANGES_REQUESTED** [review]

File: util.go (hunk 2 above)
Comment (Azer0s) [reply util.go:37]: Hmm...I don't really think `RangeOpen` is a good name for this. Maybe something like `RangeSteps` or `RangeWithSteps`? I feel like `RangeOpen` isn't really descriptive enough. 

WDYT?

Comment (jsrdxzw) [reply util.go:37]: May be RangeWithSteps is appropriate for this 😂

File: util.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package lo
+
+// Range creates an array of numbers (positive and/or negative) with given length.
+func Range(elementNum int) []int {
+	if elementNum >= 0 {
```
Comment (Azer0s) [inline util.go:5]: Range can be simplified into:

```go
func Range(start, elementNum int) []int {
	result := make([]int, If(elementNum < 0, -elementNum).Else(elementNum))
	it := If(start <= elementNum, 1).Else(-1)

	for j, i := 0, start; i != elementNum; j, i = j+1, i+it {
		result[j] = i + start
	}

	return result
}
```

File: util.go (hunk 2 above)
Comment (samber) [reply util.go:37]: RangeFromWithSteps ? 😆 

I like RangeWithSteps too

File: util.go (hunk 3 above)
Comment (jsrdxzw) [reply util.go:7]: Ogg. Yeah, I forgot there is a condition function that can be used..

**Pull request merged**